mapillary/controller/save.py

```diff
@@ -13,38 +13,125 @@
 :license: MIT LICENSE
 """
 
+# Package Imports
+import os
+import json
+import csv
+from shapely.geometry import shape
 
-def save_as_csv_controller(data: dict, path: str) -> bool:
+# Local Imports
+from utils.format import flatten_geojson
+
+
+def save_as_csv_controller(data: str, path: str, file_name: str) -> None:
     """Save data as CSV to given file path
 
     :param data: The data to save as CSV
-    :type data: dict
+    :type data: str
 
     :param path: The path to save to
     :type path: str
 
-    :return: Confirmation of whether the object was saved properly
-    :rtype: bool
-    """
-
-    # TODO: Requirement# 12A
-
-    return {"Message": "Hello, World!"}
+    :param file_name: The file name to save as
+    :type file_name: str
 
+    :return: None
+    :rtype: None
+    """
 
-def save_as_geojson_controller(data: str, path: str) -> bool:
+    # Ensure that the geojson is a dictionary
+    if isinstance(data, str):
+        data = json.loads(data)
+    
+    # Flatten the geojson
+    features = flatten_geojson(data)
+
+    try:
+        # Set the file name according to the given value. Default is "geojson.csv"
+        file_name = file_name if file_name is not None else "geojson.csv"
```

Comment:
maybe just call this mapillary.csv, or maybe even `mapillary_{timestamp_milliseconds}.csv'. Not sure if that's a good idea or not, one would overwrite the same file, one would make a new file each time, both have advantages, but of course the user can just set their own file name and path too

Reply:
We can change this to f'mapillary_{int(datetime.datetime.now().timestamp()) * 1000}.csv' which gets the current UNIX timestamp.

Reply:
Done and will be updating in the upcoming commit!